easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -553,21 +553,29 @@ def update(self, key, value, allow_duplicate=True):
         """
         Update a string configuration value with a value (i.e. append to it).
         """
+        if isinstance(value, string_type):
+            lval = [value]
+        elif isinstance(value, list):
+            lval = value
+        else:
+            msg = "Can't update configuration value for %s, because the "
+            msg += "attempted update value, '%s', is not a string or list."
+            raise EasyBuildError(msg, key, value)
+
         prev_value = self[key]
         if isinstance(prev_value, string_type):
-            if allow_duplicate or value not in prev_value:
-                self[key] = '%s %s ' % (prev_value, value)
+            for item in lval:
+                if allow_duplicate or (not re.search(r'(^|\s+)%s(\s+|$)' % re.escape(item), prev_value)):
+                    prev_value += ' %s ' % item
         elif isinstance(prev_value, list):
-            if allow_duplicate:
-                self[key] = prev_value + value
-            else:
-                for item in value:
-                    # add only those items that aren't already in the list
-                    if item not in prev_value:
-                        self[key] = prev_value + [item]
+            for item in lval:
+                if allow_duplicate or item not in prev_value:
+                    prev_value.append(item)
```

Comment:
You'll need to be careful here... `prev_value` is a reference to the original value, `.append` will append to the original value, whereas before we were creating a *new* value using `prev_value + [item]`. That was not by accident probably...

Reply:
Ah, can never remember the reference thingie...

Reply:
But it doesn't really matter, we could use self[key] directly, since the checks should be performed on the whole string/list each time.

I.e. get rid of prev_value altogether.

Reply:
We were not changing the existing value in `self[key]` inline, but we were constructing a new value and assigning `self[key]` to that value. You should keep that behavior, or we may run into surprises.

Fictic example in an easyblock:

```python
alist = ['--foo']
self.cfg['configopts'] = alist
self.cfg.update('configopts', '--bar')
for x in alist:
    ...
```

The contents of `alist` should not change (`self.cfg.update` should not have side effects)

Reply:
ah, those pescy references...